contrib/inventory/foreman.py

```diff
@@ -271,6 +290,100 @@ def to_safe(self, word):
 
     def update_cache(self, scan_only_new_hosts=False):
         """Make calls to foreman and save the output in a cache"""
+        use_inventory_report = self._use_inventory_report()
+        if use_inventory_report:
+            self._update_cache_inventory(scan_only_new_hosts)
+        else:
+            self._update_cache_host_api(scan_only_new_hosts)
+
+    def _update_cache_inventory(self, scan_only_new_hosts):
+        self.groups = dict()
+        self.hosts = dict()
+        host_data = self._post_request()
+
+        for host in host_data:
+            if host['Name'] in self.cache.keys() and scan_only_new_hosts:
+                continue
+            dns_name = host['Name']
+
+            host_params = host.get('all_parameters', {})
+            fact_data = host.get('Facts', {})
+            content_facet_data = host.get('content_facet_attributes', {})
+            content_facet_attributes = {}
```

Comment:
I'd use a dict comprehension if this can contain more than 2 elements:
```suggestion
            content_facet_attributes = {cf[0]: cf[1] for cf in content_facet_data}
```

If it's just two attributes it's easier to use the dict constructor:
```suggestion
            content_facet_attributes = dict(host.get('content_facet_attributes', {}))
```

However, the data types here are quite weird. The above line implies it's always a dict but that means the key is a tuple and the value is disregarded. It'd be good to add a test for this.